applications/tari_base_node/src/grpc/server.rs

```diff
@@ -296,6 +314,76 @@ impl base_node_grpc::base_node_server::BaseNode for BaseNodeGrpcServer {
         Ok(Response::new(rx))
     }
 
+    async fn get_new_block_template(
+        &self,
+        request: Request<base_node_grpc::PowAlgo>,
+    ) -> Result<Response<base_node_grpc::NewBlockTemplate>, Status>
+    {
+        let request = request.into_inner();
+        debug!(target: LOG_TARGET, "Incoming GRPC request for get new block template");
+        let algo = match request.pow_algo {
```

Comment:
I think you should rather implement a `From<PowAlgorithm> for u32` and a `TryFrom<u32> for PowAlgorithm` to handle these conversions. That way it is maintained in 1 clear place.

Reply:
We have one, used that one